af/af00213/src/main/java/se/inera/intyg/common/af00213/v1/rest/Af00213ModuleApiV1.java

```diff
@@ -140,7 +141,13 @@ public String createRenewalFromTemplate(CreateDraftCopyHolder draftCopyHolder, U
     @Override
     public Certificate getCertificateFromJson(String certificateAsJson) throws ModuleException {
         final var internalCertificate = getInternal(certificateAsJson);
-        return InternalToCertificate.convert(internalCertificate);
+       final var texts = getMapWithTexts(internalCertificate.getTyp(), internalCertificate.getTextVersion());
+        return InternalToCertificate.convert(internalCertificate, texts);
+    }
+
+    private SortedMap<String, String> getMapWithTexts(String type, String textVersion) {
```

Comment:
An idea. Instead of using the SortedMap directly, you could wrap it in another utility-class e.g. CertificateTextProvider. This could have a factory-method that take an IntygTexts-object and only have one method e.g. textProvider.get(FUNKTIONSNEDSATTNING_CATEGORY_TEXT).

If we do this we get more freedom to make changes if necessary and also can easily implement a situation where the key is missing we can return something instead of it being null.

Mocking the CertificateTextProvider will also become more easy that mocking a SortedMap.